### ISSUE TITLE
[CI] BooleanTermsIT testMultiValueField / testSingleValueField failing

### ISSUE THREAD

**Issue by idegtiarenko** [elastic/elasticsearch#83351]
**Build scan:**
https://gradle-enterprise.elastic.co/s/5ey776foyu7lc/tests/:server:internalClusterTest/org.elasticsearch.search.aggregations.bucket.BooleanTermsIT/testMultiValueField
https://gradle-enterprise.elastic.co/s/5ey776foyu7lc/tests/:server:internalClusterTest/org.elasticsearch.search.aggregations.bucket.BooleanTermsIT/testSingleValueField?top-execution=1

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.search.aggregations.bucket.BooleanTermsIT.testMultiValueField" -Dtests.seed=4D45D17F3171382 -Dtests.locale=nl-NL -Dtests.timezone=America/Moncton -Druntime.java=17 -Dtests.fips.enabled=true`
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.search.aggregations.bucket.BooleanTermsIT.testSingleValueField" -Dtests.seed=4D45D17F3171382 -Dtests.locale=nl-NL -Dtests.timezone=America/Moncton -Druntime.java=17 -Dtests.fips.enabled=true`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.search.aggregations.bucket.BooleanTermsIT&tests.test=testMultiValueField
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Europe/Berlin&tests.container=org.elasticsearch.search.aggregations.bucket.BooleanTermsIT&tests.sortField=FAILED&tests.test=testSingleValueField&tests.unstableOnly=true

**Failure excerpt:**
```
java.lang.ClassCastException: class org.elasticsearch.search.aggregations.bucket.terms.UnmappedTerms cannot be cast to class org.elasticsearch.search.aggregations.bucket.terms.LongTerms (org.elasticsearch.search.aggregations.bucket.terms.UnmappedTerms and org.elasticsearch.search.aggregations.bucket.terms.LongTerms are in unnamed module of loader 'app')

  at __randomizedtesting.SeedInfo.seed([4D45D17F3171382:303277C68A12E44C]:0)
  at org.elasticsearch.search.aggregations.bucket.BooleanTermsIT.testMultiValueField(BooleanTermsIT.java:117)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

**Comment by idegtiarenko** [comment]
Both tests are failing the same way. Looks like the issue was introduced in this change: https://github.com/elastic/elasticsearch/pull/82331

@nik9000 do you mind taking a look?

**Comment by nik9000** [comment]
Sorry about this folks!

**Comment by nik9000** [comment]
I think I fixed this in #81740.
